src/midend/atomics_op_lower.cpp

```diff
@@ -224,6 +261,11 @@ void graphit::AtomicsOpLower::ReduceStmtLower::visit(graphit::mir::ReduceStmt::P
         return;
     }
 
+    mir::VarExpr::Ptr var_target = mir::to<mir::VarExpr>(tensor_array_read_expr->target);
+
+    //TODO this is hack to access local array type since it is not registered in the context map
```

Comment:
I think this might be a better way to access type than accessing through mir_context

Reply:
You mean we also want to do this for global arrays too?